src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -123,12 +124,14 @@
     private PodTemplateToolLocation nodeProperties;
 
     /**
-     * @deprecated Stored as a list of yaml fragments
+     * Persisted yaml fragment
      */
-    @Deprecated
-    private transient String yaml;
+    private String yaml;
 
-    private List<String> yamls = new ArrayList<>();
+    /**
+     * List of yaml fragments used for transient pod templates. Never persisted
```

Comment:
Could this field just be deleted, and any call to `setYamls` with a list of size > 1 would just set `yaml` to something separated by `"\n----\n"` or whatever? Would be easier to follow the logic and be sure that you are not missing some complication.

Reply:
Problem I see is that would end up being an invalid yaml fragment to parse into a pod template

Reply:
As you see fit. I do not really understand what the list of YAMLs is for to begin with.